auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -241,18 +263,32 @@ public boolean equals(Object o) {
 
         IdpResponse response = (IdpResponse) o;
 
-        return (mUser == null ? response.mUser == null : mUser.equals(response.mUser))
-                && (mToken == null ? response.mToken == null : mToken.equals(response.mToken))
-                && (mSecret == null ? response.mSecret == null : mSecret.equals(response.mSecret))
-                && (mException == null ? response.mException == null : mException.equals(response.mException));
+        if (mUser != null ? !mUser.equals(response.mUser) : response.mUser != null) {
+            return false;
+        }
+        if (mToken != null ? !mToken.equals(response.mToken) : response.mToken != null) {
+            return false;
+        }
+        if (mSecret != null ? !mSecret.equals(response.mSecret) : response.mSecret != null) {
+            return false;
+        }
+        if (mException != null ? !mException.equals(response.mException) : response.mException != null) {
+            return false;
+        }
+        if (mPendingCredential != null ? !mPendingCredential.equals(response.mPendingCredential) : response.mPendingCredential != null) {
```

Comment:
To be honest I just let IntelliJ write this, and since these things are now marked `@Nullable` this is what it came up with.

But good call on the `AuthCredential` part.